index.js

```diff
@@ -19,6 +21,13 @@ class Bot extends EventEmitter {
          this.name = params.name;
 
          console.assert(params.token, 'token must be defined');
+
+         // Use a HTTP Proxy if defined in the environment. We need to accomodate both uppercase and lowercase environment definitions
+         this._agent = null;
+         if ((typeof process.env.http_proxy !== 'undefined' && process.env.http_proxy) || (typeof process.env.HTTP_PROXY !== 'undefined' && process.env.HTTP_PROXY)) {
+             this._agent = new HttpsProxyAgent(Url.parse(process.env.http_proxy || process.env.HTTP_PROXY));
```

Comment:
What will happen if Url.parse return error?

Reply:
Have updated to code handle any throws from `Url.parse` and to check if the parse didn't fail but didn't succeed either :/